### PR TITLE
Add jira writer

### PR DESCRIPTION
**Pull request by tarleb**
This adds support for Atlassian's jira markup.

Closes: #2497

### CODE REVIEW

**Comment by tarleb** [comment]
Jira's markup is a less-powerful variant of Textile. It is different enough to warrant a separate writer, but large parts were copy-pasted from the Textile writer.

There is no native math support in Jira, but plugins exist to make it work with LaTeX formulas. I'll publish a Lua filter for that somewhere.

Motivation for all this: not going crazy at work.

File: pandoc.cabal [hunk 1]
```diff
@@ -53,6 +53,7 @@ data-files:
                  data/templates/default.latex
                  data/templates/default.context
                  data/templates/default.texinfo
+                 data/templates/default.jira
```
Comment (tarleb) [inline pandoc.cabal]: I forgot to add `tables.jira`; will do tonight.

**Comment by tarleb** [comment]
Found some problems, will reopen once those are resolved.

File: src/Text/Pandoc/Writers/Jira.hs [hunk 2]
```diff
@@ -0,0 +1,322 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{- |
+   Module      : Text.Pandoc.Writers.Jira
+   Copyright   : Copyright (C) 2010-2019 John MacFarlane
```
Comment (jgm) [inline src/Text/Pandoc/Writers/Jira.hs]: Put yourself here?

**Comment by jgm** [comment]
Could you briefly describe the differences from textile, and say why it wouldn't work better to handle this by parameterizing the code in the textile writer?

**Comment by tarleb** [comment]
The main differences from Textile seem to be:

- no support for embedded HTML;
- different syntax for inline code (`@code@` vs `{{code}}`);
- different syntax for code blocks (`bc.` vs `{code}`);
- different sytnax for links (`"link example:https:://example.com` vs `{link example|https://example.com}`);
- different syntax for subscript and superscript (`[^sub^]` vs `{^sub^}`);
- different syntax for table header rows (`_.` vs `||`);
- syntax support for multi-paragraph quotes via `{quote}`;
- support for anchors (`{anchor:identifier}`);
- no way to add classes to elements;
- no support for table column alignment;
- no support for footnotes;
- no support for ordered lists starting at numbers other than 1;
- no support for definition lists.

It seems that jira syntax is only loosly based on textile; handling both by the same writer would mean littering all but the most simple element handlers with `if` statements. Despite being similar at the surface, there is not much code that can be shared by both writers.

**Comment by jgm** [comment]
OK, great. I agree it makes sense to have a separate writer.

**Comment by jgm** [comment]
@tarleb can you also update

- MANUAL.txt (list of output formats)
- the cabal description

**Comment by tarleb** [comment]
done

**Pull request merged**